.github/workflows/build-test.yml

```diff
@@ -16,6 +16,8 @@ jobs:
         os: [ubuntu-latest, windows-latest, macos-latest]
         python: ["3.8", "3.9", "3.10"]
     runs-on: ${{ matrix.os }}
+    env:
```

Comment:
you should be able to set it on the workflow level like here https://docs.github.com/en/actions/learn-github-actions/environment-variables

and I think it needs to be `agg`: https://matplotlib.org/stable/users/explain/backends.html

Reply:
I put it in this job because it's the only one that uses matplotlib right now, but I agree setting it on the workflow level would be more future proof. 

On the page you sent me it says in a note: "The names of builtin backends case-insensitive; e.g., 'QtAgg' and 'qtagg' are equivalent." Happy to change it into `agg` though if you like that better.